version: '3.9'

services:
  django-app:
    build:
      context: .
      dockerfile: deploy/Dockerfile.django.prod
    env_file:
      - .env
    environment:
      ENVIRONMENT: PROD
    ports:
      - 8000:8000
    volumes:
      - .:/app
    depends_on:
      db:
        condition: service_healthy
    network_mode: host


  db:
    image: postgres:latest
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_DATABASE}
    volumes:
      - postgres-data:/var/lib/postgresql/data
    ports:
      - "${DB_PORT}:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USER} && PGPASSWORD=${DB_PASSWORD} psql -U ${DB_USER} -d ${DB_DATABASE} -c 'SELECT 1'"]
      interval: 10s
      timeout: 3s
      retries: 3
    network_mode: host


volumes:
  postgres-data:
